autopilot/contractor.go

```diff
@@ -443,49 +460,53 @@ func (c *contractor) runContractRenewals(cfg api.AutopilotConfig, blockHeight, c
 		)
 	}()
 
-	// perform renewals first
-	for i, ci := range append(toRenew, toRefresh...) {
-		isRefresh := i >= len(toRenew)
+	for _, ci := range toRenew {
+		// TODO: keep track of consecutive failures and break at some point
 
 		// break if the contractor was stopped
 		if c.isStopped() {
 			break
 		}
 
+		// convenience variables
+		contract := ci.contract
+		settings := ci.settings
+		fcid := contract.ID
+		rev := contract.Revision
+		hk := contract.HostKey()
+		fs := contract.FileSize()
+
 		// calculate the renter funds
-		renterFunds, err := c.renterFundsEstimate(cfg, currentPeriod, blockHeight, ci, isRefresh)
+		renterFunds, err := c.renewFundingEstimate(cfg, blockHeight, ci)
 		if err != nil {
-			c.logger.Errorw(
-				fmt.Sprintf("could not get refresh funding estimate, err: %v", err),
-				"hk", ci.contract.HostKey(),
-				"fcid", ci.contract.ID,
-				"refresh", isRefresh,
-			)
+			c.logger.Errorw(fmt.Sprintf("could not get renew funding estimate, err: %v", err), "hk", hk, "fcid", fcid)
 			continue
 		}
 
 		// check our budget
 		if budget.Cmp(renterFunds) < 0 {
-			c.logger.Debugw(
-				"insufficient budget",
-				"budget", budget,
-				"needed", renterFunds,
-				"renew", !isRefresh,
-				"refresh", isRefresh,
-			)
+			c.logger.Debugw("insufficient budget", "budget", budget, "needed", renterFunds)
 			break
 		}
 
-		// derive the renter key
-		newRevision, err := c.renewContract(cfg, currentPeriod, ci.contract, renterAddress, renterFunds, isRefresh)
+		// calculate the host collateral
+		endHeight := endHeight(cfg, c.currentPeriod())
+		timeExtension := endHeight - rev.WindowEnd
+		hostCollateral := renewContractCollateral(settings, renterFunds, fs, timeExtension)
+
+		// check if the added collateral is below the threshold
+		basePrice, baseCollateral := rhpv2.ContractBaseCosts(rev.FileContract, settings, endHeight)
+		_, hostMissedPayout, _ := rhpv2.CalculatePayouts(basePrice, baseCollateral, hostCollateral, settings)
+		if isBelowCollateralThreshold(cfg, settings, hostMissedPayout) {
```

Comment:
I think we should only fail a refresh in this case. If we fail the renew, we might lose read-access to the data too before we can migrate away from the host.

Reply:
Yeah wasn't sure but it makes sense, I updated it.